ls1intum/Artemis
# Add more customization options for test case weights in programming exercises

This PR adds more customization options for test case weights in programming exercises: `bonus_multiplier`, `bonus_points` and floating values for weights.

### Checklist

### Motivation and Context
The current customization options for test case weights in programming exercises have limitations.
We can now use floating values for weights and can additionally define bonus multiplier and bonus points for individual tests to give more points than just calculated by the weight.

### Description
* Add two new attributes `bonus_multiplier` and `bonus_points`.
* Change type of `weight` to `double`
* Add `bonus_points` to Exercise (currently unused)

### Steps for Testing
1. Participate in programming exercises
2. Update the test case entries (try out different combinations of weight, bonus multiplier, bonus points)
3. Run the tests again (or push changes again)
4. Check that the score updates accordingly

The formula for each successful test is `weight` * `bonus_multiplier` + `bonus_points`.
The sum of successful tests will be divided by the sum of all test weights.

### Screenshots
<img width="1153" alt="image" src="https://user-images.githubusercontent.com/744067/89535114-3dfae180-d7f6-11ea-8ee1-7366434099a7.png">
